keybot/darwinbot.go

```diff
@@ -59,11 +60,11 @@ func (d *darwinbot) Run(bot slackbot.Bot, channel string, args []string) (string
 		smokeTest := true
 		skipCI := *buildDarwinSkipCI
 		testBuild := *buildDarwinTest
-		// Don't smoke, wait for CI or promote test if custom build
-		if *buildDarwinCientCommit != "" || *buildDarwinKbfsCommit != "" {
-			smokeTest = false
-			skipCI = true
-			testBuild = true
+		// If it's a custom build, make it a test build unless --smoke is passed.
+		if *buildDarwinClientCommit != "" || *buildDarwinKbfsCommit != "" {
+			smokeTest = *buildDarwinSmoke
```

Comment:
so you can only turn off smoke test if a commit is specified? Seems simpler to just honor each flag regardless - the winbot needs some cleanup like this too.

Reply:
Decided against this because it would need retraining (of both humans and bots) to pass --smoke (or --no-smoke) where it isn't passed today.  I agree a cleanup would be nice, though.